src/sentry/integrations/client.py

```diff
@@ -194,26 +210,37 @@ def _request(
                 sample_rate=1.0,
                 tags={"integration": self.integration_name, "status": "connection_error"},
             )
+            log_request_failure(
+                self.integration_name, "connection_error", e, context=self.logging_context
+            )
             raise ApiHostError.from_exception(e)
         except Timeout as e:
             metrics.incr(
                 "integrations.http_response",
                 sample_rate=1.0,
                 tags={"integration": self.integration_name, "status": "timeout"},
             )
+            log_request_failure(self.integration_name, "timeout", e, context=self.logging_context)
             raise ApiTimeoutError.from_exception(e)
         except HTTPError as e:
             resp = e.response
             if resp is None:
                 track_response_code(self.integration_name, "unknown")
+                log_request_failure(
+                    self.integration_name, "unknown", e, context=self.logging_context
+                )
                 self.logger.exception(
                     "request.error", extra={"integration": self.integration_name, "url": full_url}
                 )
                 raise ApiError("Internal Error")
             track_response_code(self.integration_name, resp.status_code)
+            log_request_failure(
+                self.integration_name, resp.status_code, e, context=self.logging_context
+            )
             raise ApiError.from_response(resp)
 
         track_response_code(self.integration_name, resp.status_code)
+        log_request_failure(self.integration_name, resp.status_code, context=self.logging_context)
```

Comment:
@ceorourke Seeing as how we are always calling `metrics.incr`  along with `log_request_failure`, I think it makes sense to put them both in a new function that both logs the error and send the metric to Datadog. As it is, it's just a lot of repeating code.